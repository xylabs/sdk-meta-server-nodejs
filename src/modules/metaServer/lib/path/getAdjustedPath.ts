import Path from 'node:path'

import { Request } from 'express'

import { isKnownFileExtension } from '../file/index.js'

export const getAdjustedPath = (req: Request): string => {
  if (isKnownFileExtension(req.path)) return req.path
  return Path.join(req.path, 'index.html')
}
