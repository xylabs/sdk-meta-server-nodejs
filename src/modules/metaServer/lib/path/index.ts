export * from './getAdjustedPath'
export * from './isHtmlLike'
