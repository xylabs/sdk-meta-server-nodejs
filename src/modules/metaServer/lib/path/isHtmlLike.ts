import Path from 'node:path'

import { Request } from 'express'

import { isKnownFileExtension } from '../file/index.js'

/**
 * Returns true request path could be referring
 * to an HTML file
 * @param req The web request
 * @returns True if the request path could be referring,
 * to an HTML document, false otherwise
 */
export const isHtmlLike = (req: Request): boolean => {
  const ext = Path.extname(req.path)
  if (ext.length === 0) return true
  if (ext.toLowerCase() === '.html') return true
  if (!isKnownFileExtension(req.path)) return true
  return false
}
