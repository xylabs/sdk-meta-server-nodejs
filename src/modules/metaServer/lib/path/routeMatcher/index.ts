export * from './createGlobMatcher.js'
export * from './createRegexMatcher.js'
export * from './RouteMatcher.js'
