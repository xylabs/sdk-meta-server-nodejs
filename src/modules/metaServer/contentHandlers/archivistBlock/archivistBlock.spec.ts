import { tryParseInt } from '@xylabs/sdk-api-express-ecs'
import { readFile } from 'fs/promises'
import { Server } from 'http'
import { join } from 'path'
import { SuperTest, Test } from 'supertest'

import { getServerOnPort } from '../../spec'

const payloadUris = [
  '/archive/temp/block/hash/eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e?network=kerplunk',
  '/archive/temp/payload/hash/2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99?network=kerplunk',
]

const testServerPort = 12345

describe('archivistBlock', () => {
  let server: Server
  let agent: SuperTest<Test>
  beforeAll(() => {
    jest.spyOn(console, 'log').mockImplementation(() => {
      // Disable `console.log` from real server starting up for these tests
    })
    const baseDir = __dirname
    ;[server, agent] = getServerOnPort(testServerPort, baseDir)
  })
  afterAll((done) => {
    server.close(done)
  })
  it.each(payloadUris)('Modifies the head with block information', async (payloadUri) => {
    // Get this file via server
    const response = await agent.get(payloadUri).expect(200)
    expect(response).toBeTruthy()

    // Validate HTML headers
    const headers = response.headers
    expect(headers).toBeTruthy()
    expect(headers['content-type']).toBeTruthy()
    expect(headers['content-type']).toBe('text/html; charset=utf-8')

    expect(headers['content-length']).toBeTruthy()
    const contentLength = tryParseInt(headers['content-length'])
    expect(contentLength).toBeDefined()
    // Read file in original HTML file
    const originalFile = await readFile(join(__dirname, 'index.html'))
    // HTTP Content-Length header should reflect that modified file is
    // bigger than original after modification
    expect(contentLength).toBeGreaterThan(originalFile.byteLength)

    // Validate HTML document
    const actual = response.text
    expect(actual).toBeTruthy()
    expect(actual).toMatchSnapshot()
  })
})
