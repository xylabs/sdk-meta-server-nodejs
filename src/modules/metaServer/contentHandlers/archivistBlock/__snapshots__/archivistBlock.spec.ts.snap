// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`archivistBlock Modifies the head with block information 1`] = `
"<!DOCTYPE html><html lang="en"><head>
  <meta charset="utf-8">
  <title>XYO 2.0: Bound Witness | eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e</title>
<meta property="og:title" content="XYO 2.0: Bound Witness | eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e"><meta property="og:url" content="http://127.0.0.1:12345/archive/temp/block/hash/eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e?network=kerplunk"><meta property="twitter:title" content="XYO 2.0: Bound Witness | eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e"><meta property="description" content="A XYO 2.0 network.xyo.boundwitness payload with the hash eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e."><meta property="title" content="XYO 2.0: Bound Witness | eb27161e9d12403a0a2c49590ca27c215c70382fa5f8aa7d06a28ed24394748e"></head>

<body>
  <pre>Test</pre>


</body></html>"
`;

exports[`archivistBlock Modifies the head with block information 2`] = `
"<!DOCTYPE html><html lang="en"><head>
  <meta charset="utf-8">
  <title>XYO 2.0: Payload | 2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99</title>
<meta property="og:title" content="XYO 2.0: Payload | 2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99"><meta property="og:url" content="http://127.0.0.1:12345/archive/temp/payload/hash/2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99?network=kerplunk"><meta property="twitter:title" content="XYO 2.0: Payload | 2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99"><meta property="description" content="A XYO 2.0 network.xyo.location payload with the hash 2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99."><meta property="title" content="XYO 2.0: Payload | 2096d4e1a3c0bf1ead5e7b2144bf98e39d0679c343d79c896a0d836479475e99"></head>

<body>
  <pre>Test</pre>


</body></html>"
`;
